packages/devtools_app/lib/src/shared/config_specific/copy_to_clipboard/copy_to_clipboard.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be found
+// in the LICENSE file.
+
+import 'package:flutter/services.dart';
+import 'package:logging/logging.dart';
+
+import '../../globals.dart';
+import '_copy_to_clipboard_stub.dart'
+    if (dart.library.js_interop) '_copy_to_clipboard_web.dart'
+    if (dart.library.io) '_copy_to_clipboard_desktop.dart';
+
+final _log = Logger('copy_to_clipboard');
+
+/// Attempts to copy a String of `data` to the clipboard.
+///
+/// Shows a `successMessage` [Notification] on the passed in `context`, if the
+/// copy is successfully done using the [Clipboard.setData] api. Otherwise it
+/// attempts to post the [data] to the parent frame where the parent frame will
+/// try to complete the copy (this fallback will only work in VSCode).
+Future<void> copyToClipboard(
+  String data,
+  String? successMessage,
+) async {
+  try {
+    await Clipboard.setData(
+      ClipboardData(
+        text: data,
+      ),
+    );
+
+    if (successMessage != null) notificationService.push(successMessage);
+  } catch (e) {
+    _log.warning(
```

Comment:
check if we are embedded and in VS code before doing any of this

Reply:
You can check if we are embedded by using `ideTheme.embed` and you can check which IDE is hosting by checking here: https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/shared/analytics/_analytics_web.dart/#L733.

Although, I think we may want to modify the IdeTheme class to include the IDE name so we can look it up from there instead of having to look at our analytics file.